mesonbuild/modules/qt.py

```diff
@@ -373,6 +374,89 @@ def _compile_resources_impl(self, state: 'ModuleState', kwargs: 'ResourceCompile
 
         return targets
 
+    @FeatureNew('qt.compile_qwayland', '1.0')
+    @noPosargs
+    @typed_kwargs(
+        'qt.compile_qwayland',
+        KwargInfo('client', bool, default=True),
+        KwargInfo('server', bool, default=False),
+        KwargInfo('source', (str, File)),
+        KwargInfo('extra_args', ContainerTypeInfo(list, str), listify=True, default=[]),
+        KwargInfo('method', str, default='auto')
```

Comment:
Perhaps it would make sense to share kwargs with their wayland module equivalent